pinot-core/src/test/java/com/linkedin/pinot/core/realtime/RealtimeFileBasedReaderTest.java

```diff
@@ -101,13 +98,30 @@ public static void before() throws Exception {
 
     RealtimeSegmentConverter conveter =
         new RealtimeSegmentConverter(realtimeSegment, "/tmp/realtime", schema, tableName, segmentName, null);
-    conveter.build();
+    conveter.build(segmentVersion);
 
     offlineSegment = Loaders.IndexSegment.load(new File("/tmp/realtime").listFiles()[0], ReadMode.mmap);
   }
 
   @Test
-  public void testDataSourceWithoutPredicateForSingleValueDimensionColumns() {
+  public void allTestsV1() throws Exception {
+    setUp(SegmentVersion.v1);
+    testDataSourceWithoutPredicateForSingleValueDimensionColumns();
+    testDataSourceWithoutPredicateForSingleValueMetricColumns();
+    testDataSourceWithoutPredicateForSingleValueTimeColumns();
+    testDataSourceWithoutPredicateForMultiValueDimensionColumns();
+  }
+
+  @Test
```

Comment:
Since this is an enum, I cannot make it take a random string. In the code, I use Segment.fromStringOrDefault() method. Would u be fine if I just add a test for this method?